Create international-localization-localization.md (#181)
PrestaShop/prestashop-specs: back-office/international/localization/international-localization-localization.md

**Download pack data**

The _Download pack data_ toggle displays a YES/NO toggle button which is set to 'YES' by default. If 'YES' is set, the localizations that are selected on 'Content to import' section will be downloaded from prestashop.com CLDR directory and override the local CLDR localizations. If set to 'NO' it will use the local CLDR localizations (localization/CLDR/core/common/main/country-code.xml, localization/CLDR/core/common/main/country-code_COUNTRY-ISO-CODE.xml). The local CLDR localizations can be modified manually by modifying the local files mentioned.

After any modifications applied to local CLDR files, the PrestaShop cache (back-office/advanced-parameters/performance/clear-cache.md) must be cleared to be shown.

**Import**

Pressing the _Import_ button applies the 'Import a localization pack' options. Clicking on it might affect both FrontOffice and BackOffice or change the default price behaviors based on options chose from 'Content to import' and 'Download pack data' fields.

> 

Reviewer: ```suggestion

#### Multistore behavior

States will be imported for all shops, no matter the context selected.
Taxes will be imported for all shops, no matter the context selected. Taxes rules will be imported for the selected shop(s) in the context.
Currencies will be imported for the selected shop(s) in the context.
Languages will be imported for all shops, no matter the context selected.

```

### Configuration

Using this section will give the user the ability to change default localization configuration such as default langiage, country, and currency. Changing this configuration will override the default behaviours.

**Default language**

> The _Default language_ droplist displays the languages that are already imported to the shop by user from 'Content to import' section. By default it shows the language that user defined on PrestaShop installation process. Changing it will change the language that is selected by default on FrontOffice (front-office).

Author: Yes, of course. I've written that by default it's showing the language that is defined on installation process. But if needed, we can add a note about default BO lang as well.

> The _Default language_ droplist displays the languages that are already imported to the shop by user from 'Content to import' section. By default it shows the language that user defined on PrestaShop installation process. Changing it will change the language that is selected by default on FrontOffice (front-office).

Reviewer 2: It'd be great, thank you very much!

The _Set language from browser_ toggle displays a 'YES/NO' toggle button which is set to 'YES' by default. If 'YES' is set, the default language will be changed based on user's default browser langauage automatically on FrontOffice (front-office).

**Default country**

The _Default country_ droplist displays the full list of all countries. The user can select the default country selected on FrontOffice (front-office) from droplist or typing at least one word using the search field on the top of droplist.

**Set default country from browser language**

The _Set default country from browser language_ toggle displays a 'YES/NO' toggle button which is set to 'YES' by default. If 'YES' is set, the default country on FrontOffice (front-office) will be selected automatically based on user's default browser language.

**Default currency**

The _Default currency_ droplist displays the currencies that are already imported to the shop by user from 'Content to import' section. By default, it shows the currency based on default country selected on PrestaShop process. Changing it will change the currency that is selected by default on FrontOffice (front-office).

**Time zone**

> The _Time zone_ droplist displays the full list of all standard time zones. The user can select the shop's default Time zone from droplist or typing at least one word using the search field on the top of droplist.

Reviewer 2: I wonder if we shouldn't add the GTM/UTC information along with the timezone name.

**Save**

> Pressing the _Save_ button applies the 'Configuration' options. Clicking on it might affect both FrontOffice and BackOffice or change the default localization configuration based on options chose from 'Configuration' fields.

Reviewer 2: Currently, modifying the default language doesn't seem to have an impact on the 'Language' selected in the employee's profile. Do you think it should?

### Local units

Using this section will able the user to change the default local units such as weight, distance, volume, and dimension.

**Weight unit**

> The _Weight unit_ provides a text field for the user to change default weight unit on shop. The user can type any local unit for weight such as kg for kilograms, g for grams or unit in their own local country/language.

Reviewer 2: ```suggestion
The _Weight unit_ provides a text field for the user to change the default weight unit for the shop. The user can type any local unit for weight such as 'kg' for kilograms, 'g' for grams, or unit in their own local country/language.
```

**Distance unit**

> The _Distance unit_ provides a text field for the user to change default distance unit on shop. The user can type any local unit for distance such as km for kilometers, mi for miles or unit in their own local country/language.

Reviewer 2: ```suggestion
The _Distance unit_ provides a text field for the user to change the default distance unit for the shop. The user can type any local unit for distance such as 'km' for kilometers, 'mi' for miles, or unit in their own local country/language.
```

**Volume unit**

> The _Volume unit_ provides a text field for the user to change default volume unit on shop. The user can type any local unit for volume such as L for liter, gal for galon or unit in their own local country/language.

Reviewer 2: ```suggestion
The _Volume unit_ provides a text field for the user to change the default volume unit for the shop. The user can type any local unit for volume such as 'L' for liter, 'gal' for gallon, or unit in their own local country/language.
```

**Dimension unit**

> The Dimension unit_ provides a text field for the user to change default dimension unit on shop. The user can type any local unit for dimentsion such as cm for centimeters, in for inches or unit in their own local country/language.

Reviewer 2: ```suggestion
The Dimension unit_ provides a text field for the user to change the default dimension unit for the shop. The user can type any local unit for dimension such as 'cm' for centimeters, 'in' for inches, or unit in their own local country/language.
```

**Save**

Pressing the _Save_ button applies the 'Local units' options. Clicking on it might affect both FrontOffice and BackOffice or change the default units based on fields filled from 'Local units' section.

### Advanced

Using this section will define the local country and its language where the server is located.

**Language identifier**

> The _Language identifier_ field allows user to set the local language of the country that their server is located. This field must be filled with language iso code in lowercase using the 'ISO 639-1' standard.

Reviewer 2: ```suggestion
The _Language identifier_ field allows the user to set the local language of the country where the server is located. This field must be filled with the language ISO code in lowercase using the 'ISO 639-1' standard.
```

Reviewer 2: What if the user has a server located in Germany but his business is officially located in France? Shouldn't we rather ask for his country's language?

Author: As I know, this field defines the "Server's location" not their office location. The office information can be filled on "Shop Parameters > Contacts > Stores".

Reviewer 2: @mehrshadz, thanks for the reply, I think I miss the point of having the language and the country identifiers of the server's localization... what is the purpose?

Author: To be honest, it's better to get your answer from @PrestaShop/product-team 😄

Reviewer 2: @marionf, @MatShir, @eternoendless, does anyone has an idea?

Reviewer: I looked at the documentation, it is indeed the server's local language and country, but I do not know what is the purpose

Author: It's been near 9 years that I'm using PrestaShop and I've never noticed this option! 😆 
I took a deeper look into this option, it uses a class called AdvancedConfiguration inside src/Adapter/Localization/AdvancedConfiguration.php file. This class is used in some files:
1. vendor/composer/autoload_classmap.php - Created by Composer
2. vendor/composer/autoload_static.php - Created by Composer
3. src/PrestaShopBundle/Form/Admin/Improve/International/Localization/LocalizationFormDataProvider.php - inside the LocalizationFormDataProvider class which is responsible for 'Improve > International > Localization' page form data handling
4. A file called getPrestaShop_Adapter_LocalizationAdvanced_ConfigurationService.php inside var/cache/dev folder that is generated by Symfony Dependency Injection Component.

My most probable guess is that these two "server language and country identifier" fields are used to set Symfony Localization configuration.

If I'm wrong, I'll be glad to know more about this option ❤️

Reviewer 3: I looked it up, the language identifier and the country identified are tied to the `PS_LOCALE_LANGUAGE` and `PS_LOCALE_COUNTRY` settings respectively.

It has nothing to do with Symfony, they go waaaay back, at least as far as 1.4. 

They are used in [config.inc.php](https://github.com/PrestaShop/PrestaShop/blob/1.7.7.0/config/config.inc.php#L154-L159) to initialize PHP's localization settings via [setlocale()](https://www.php.net/manual/en/function.setlocale.php). These govern default timezone, date formats, and so on when using localization aware functions. Most places in PrestaShop aren't affected by this because we use our own CLDR library, but a default must be set, or PHP will complain about it. Since these functions use IETF langage tags (eg. 'en_US') and not language/countries, this could be improved to a single field that containing the IETF language tag.

Reviewer 2: Hej, thanks for the complete reply! So, to summarize: we don't really need those two fields anymore but the IETF language tag is required by PHP so it'd make more sense to turn them into a unique field, named `IETF language tag`, automatically filled with the right data (being the language tag of the default language of the store).

I'm wondering if users should be able to change this language tag... If we should keep this field editable, the user must have the possibility to do so by modifying the field and clicking 'Save'. I bet IETF language tags won't be very meaningful so I suggest adding a help text to let them know how to pick the proper IETF tag (combination of the 639 and 3166-1 ISO codes).

I'll investigate further to know exactly what are the elements impacted by this localization setting. 🤓

Author: @eternoendless Finally, someone took us out of darkness 💡 Thanks Pablo for your complete explanation.
I agree with @LouiseBonnard I think it's better to add a help text, or fill this option with a default non-editable value if we're using CLDR and it's only used for PHP configuration. Of course, Pablo's suggestion on this would be really helpful.

Reviewer 2: From what I've explored, no date nor price format (BO & FO) looks impacted by this feature.

Author: @LouiseBonnard @eternoendless 
Shall we mark this conversation as resolved?

Reviewer 2: I prefer to leave it open: from a functional point of view, no feature looks impacted but as @eternoendless stated, it might be required for the PHP configuration. I'm in favor of investigating a bit more the technical side of this information before making up our mind.

@PrestaShop/prestashop-core-developers, if someone can have a look at it, I'm available to talk about it and finally complete those specifications. :-)

**Country identifier**

> The _Country identifier_ field allows user to set the local country where their server is located. This field must be filled with country iso code in lowercase using the 'ISO 3166-1 alpha-2' standard.

Reviewer 2: ```suggestion
The _Country identifier_ field allows the user to set the local country where the server is located. This field must be filled with the country ISO code in lowercase using the 'ISO 3166-1 alpha-2' standard.
```

Reviewer 2: Same question as for the language. ;-)

Author: The same answer :D

**Localization pack you want to import**

The list of countries will show up on clicking on it. It can be selected by:
- Searching the list via the search input field by typing the complete country name or part of it (at least one word).
- Scrolling down the list and find the country which are sorted by their name (a-z).

**Content to import**

The contents to import are listed as check-boxes to be selected:
- States (checked by default)
- Taxes (checked by default)
- Currencies (checked by default)
- Languages (checked by default)
- Units (e.g. weight, volume, distance) (checked by default)
- Change the behavior of the price display for groups (un-cheked by default)
By checking the 'Change the behavior of the price display for groups' the behaviour of displaying prices will be affected on both BackOffice (back-office/international/locations/Countries.md, back-office/shop-parameters/customer-settings/groups.md) and FrontOffice (front-office/product-page.md, front-office/products/list.md, front-office/Discounts). This option will override the default behaviour for selected country if you already have it inside your shop.

**Download pack data**

> Shows a YES/NO toggle button which is set to 'YES' by default. If 'YES' is set, the localizations that are selected on 'Content to import' section will be downloaded from prestashop.com CLDR directory and override the local CLDR localizations. If set to 'NO' it will use the local CLDR localizations (localization/CLDR/core/common/main/country-code.xml, localization/CLDR/core/common/main/country-code_COUNTRY-ISO-CODE.xml). The local CLDR localizations can be modified manually by modifying the local files mentioned.</br>

Reviewer 4: ```suggestion
The _Download pack data_ toggle displays a YES/NO toggle button which is set to 'YES' by default. If 'YES' is set, the localizations that are selected on 'Content to import' section will be downloaded from prestashop.com CLDR directory and override the local CLDR localizations. If set to 'NO' it will use the local CLDR localizations (localization/CLDR/core/common/main/country-code.xml, localization/CLDR/core/common/main/country-code_COUNTRY-ISO-CODE.xml). The local CLDR localizations can be modified manually by modifying the local files mentioned.

```

After any modifications applied on local CLDR files, the PrestaShop cache (back-office/advanced-parameters/performance/clear-cache.md) must be cleared to be shown.

**Import**

> The import button is shown to apply the 'Import a localization pack' options. Clicking on it might affect both FrontOffice and BackOffice or change the default price behaviours based on options chose from 'Content to import' and 'Download pack data' fields.

Reviewer 4: ```suggestion
Pressing the _Import_ button applies the 'Import a localization pack' options. Clicking on it might affect both FrontOffice and BackOffice or change the default price behaviors based on options chose from 'Content to import' and 'Download pack data' fields.
```
